experimental/rsocket-src/transports/TcpConnectionFactory.cpp

```diff
@@ -1,99 +1,84 @@
 // Copyright 2004-present Facebook. All Rights Reserved.
 
 #include "rsocket/transports/TcpConnectionFactory.h"
+
 #include <folly/io/async/AsyncSocket.h>
 #include <folly/io/async/EventBaseManager.h>
+#include <glog/logging.h>
+
 #include "src/framed/FramedDuplexConnection.h"
 #include "src/tcp/TcpDuplexConnection.h"
 
 using namespace reactivesocket;
-using namespace folly;
 
 namespace rsocket {
 
-// create new ScopedEventBaseThread
-// create new EventBase from it
-// create new AsyncSocket
-// connect
-// create FramedDuplexConnection
-// TODO create variant that takes an existing EventBase
-class SocketConnectorAndCallback : public AsyncSocket::ConnectCallback {
+namespace {
+
+class ConnectCallback : public folly::AsyncSocket::ConnectCallback {
  public:
-  SocketConnectorAndCallback(
-      OnConnect onConnect,
-      SocketAddress addr,
-      EventBase* eventBase)
-      : addr_(addr), onConnect_(onConnect), eventBase_(eventBase) {}
-
-  ~SocketConnectorAndCallback() {
-    LOG(INFO) << "SocketConnectorAndCallback => destroy";
-  }
+  ConnectCallback(folly::SocketAddress address, OnConnect onConnect)
+      : address_(address), onConnect_{std::move(onConnect)} {
+    VLOG(2) << "Constructing ConnectCallback";
 
-  void connect() {
-    // now start the connection asynchronously
-    eventBase_->runInEventBaseThreadAndWait([this]() {
-      LOG(INFO) << "ConnectionFactory => starting socket";
-      socket_.reset(new folly::AsyncSocket(eventBase_));
+    // Set up by ScopedEventBaseThread.
+    auto evb = folly::EventBaseManager::get()->getExistingEventBase();
+    DCHECK(evb);
 
-      LOG(INFO) << "ConnectionFactory => attempting connection to "
-                << addr_.describe() << std::endl;
+    VLOG(3) << "Starting socket";
+    socket_.reset(new folly::AsyncSocket(evb));
 
-      socket_->connect(this, addr_);
+    VLOG(3) << "Attempting connection to " << address_;
 
-      LOG(INFO) << "ConnectionFactory  => DONE connect";
-    });
+    socket_->connect(this, address_);
   }
 
- private:
-  SocketAddress addr_;
-  folly::AsyncSocket::UniquePtr socket_;
-  OnConnect onConnect_;
-  EventBase* eventBase_;
+  ~ConnectCallback() {
+    VLOG(2) << "Destroying ConnectCallback";
+  }
 
   void connectSuccess() noexcept {
-    LOG(INFO) << "ConnectionFactory => socketCallback => Success";
+    std::unique_ptr<ConnectCallback> deleter(this);
```

Comment:
I'm not sure it is, doing `delete this;` in folly socket callbacks is a common thing.  Most of the time I see people doing it with `SCOPE_EXIT { delete this; };`, but the `unique_ptr` way also works.